sycl/source/detail/queue_impl.cpp

```diff
@@ -86,6 +91,79 @@ void queue_impl::addEvent(event Event) {
   MEvents.push_back(std::move(Event));
 }
 
+void queue_impl::wait(const cl::sycl::detail::code_location &CodeLoc) {
+#ifdef XPTI_ENABLE_INSTRUMENTATION
+  xpti::trace_event_data_t *TraceEvent = nullptr;
+  std::string Name;
+  int32_t StreamID = xptiRegisterStream(SYCL_STREAM_NAME);
+  if (xptiTraceEnabled()) {
+    xpti::payload_t p;
+    bool hasSourceInfo = false;
+    // We try to create a unique string for the wait() call
+    // by combining it with the queue address
+    xpti::utils::StringHelper NG;
+    Name = NG.nameWithAddress("queue.wait", (void *)this);
+
+    std::string Check(CodeLoc.fileName());
+    if (Check.find("unknown") == std::string::npos) {
+      // We have souce code location information
+      p = xpti::payload_t(Name.c_str(), CodeLoc.fileName(),
+                          CodeLoc.lineNumber(), CodeLoc.columnNumber(),
```

Comment:
Please, move profiling related code to separate functions to clearly separate profiling from other code.

Reply:
Done!